### PR TITLE
Fix vapp description and add vapp.Rename

### PR DESCRIPTION
**Pull request by dataclouder**
* Added parameter `description` to method `vdc.ComposeRawVapp` (This method is used in `terraform-provider-vcd` to create a vApp.)
* Added methods `vapp.Rename`, `vapp.UpdateDescription`, `vapp.UpdateNameDescription`. This make it possible to update the vApp's name and description in Terraform.

### CODE REVIEW

File: govcd/common_test.go [hunk 1]
```diff
@@ -672,7 +672,7 @@ func deleteVapp(vcd *TestVCD, name string) error {
 // makeEmptyVapp creates a given vApp without any VM
 func makeEmptyVapp(vdc *Vdc, name string) (*VApp, error) {
 
-	err := vdc.ComposeRawVApp(name)
+	err := vdc.ComposeRawVApp(name, name)
```
Comment (lvirbalas) [inline govcd/common_test.go:675]: Should the second parameter hold desription?

File: govcd/vapp.go [hunk 2]
```diff
@@ -1375,3 +1376,71 @@ func (vapp *VApp) getOrgInfo() (orgInfoType, error) {
...
+
+// UpdateNameDescription can change the name and the description of a vApp
+// If name or description are empty, they are not changed.
+func (vapp *VApp) UpdateNameDescription(newName, newDescription string) error {
+	if vapp == nil || vapp.VApp.HREF == "" {
+		return fmt.Errorf("vapp or href cannot be empty")
+	}
+
+	// Skip update if we are using the original values
+	if (newName == vapp.VApp.Name || newName == "") && (newDescription == vapp.VApp.Description || newDescription == "") {
```
Comment (lvirbalas) [inline govcd/vapp.go:1388]: How would one set description to empty?

File: govcd/vapp_test.go [hunk 3]
```diff
@@ -1557,3 +1558,101 @@ func (vcd *TestVCD) Test_AddNewVMWithComputeCapacity(check *C) {
...
+	AddToCleanupList(vappName, "vapp", "", "Test_RenameVapp")
+
+	check.Assert(vapp.VApp.Name, Equals, vappName)
+	check.Assert(vapp.VApp.Description, Equals, vappName)
+
+	// Need a slight delay for the vApp to get the links that are needed for renaming
+	time.Sleep(time.Second)
+
+	// change description
+	vcd.testUpdateVapp(check, vapp, "", newVappDescription, nil)
```
Comment (lvirbalas) [inline govcd/vapp_test.go:1628]: I may have missed it, but we need a test which tries to re-set description to empty.

File: govcd/common_test.go (hunk 1 above)
Comment (dataclouder) [reply govcd/common_test.go:675]: This is a testing function that is not using the description. Since I changed the signature of `ComposeRawVapp`, I added the vApp name as description.

File: govcd/vapp.go (hunk 2 above)
Comment (dataclouder) [reply govcd/vapp.go]: Changed to allow description removal

File: govcd/vapp_test.go (hunk 3 above)
Comment (dataclouder) [reply govcd/vapp_test.go]: added

File: govcd/common_test.go (hunk 1 above)
Comment (lvirbalas) [reply govcd/common_test.go:675]: Nit, but you could use something like `"Description of " + name` - this way in the unlikely case a person would look into this populated field, it would be clear that it works and is not a bug.

Comment (dataclouder) [reply govcd/common_test.go]: Changed.
However, the life span of a vApp created by this function is usually measured in minutes. The likelihood of someone looking into it is negligible.

Comment (lvirbalas) [reply govcd/common_test.go]: That's why it's a nit :) BTW, I think gofmt is needed there.

**Review by lvirbalas — APPROVED** [review]
LGTM!

File: govcd/vapp.go [hunk 4]
```diff
@@ -1375,3 +1376,71 @@ func (vapp *VApp) getOrgInfo() (orgInfoType, error) {
...
+		return nil
+	}
+
+	opType := "application/vnd.vmware.vcloud.recomposeVAppParams+xml"
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply govcd/vapp.go]: Yes. Fixed

File: govcd/vapp.go [hunk 5]
```diff
@@ -1375,3 +1376,71 @@ func (vapp *VApp) getOrgInfo() (orgInfoType, error) {
 	}
 	return getOrgInfo(vapp.client, vdc.Vdc.Link, vapp.VApp.ID, vapp.VApp.Name, "vApp")
 }
+
+// UpdateNameDescription can change the name and the description of a vApp
+// If name or description are empty, they are not changed.
+func (vapp *VApp) UpdateNameDescription(newName, newDescription string) error {
+	if vapp == nil || vapp.VApp.HREF == "" {
+		return fmt.Errorf("vapp or href cannot be empty")
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply govcd/vapp.go]: changed

**Review by Didainius — APPROVED** [review]
It works!
